R/assertions.R

```diff
@@ -55,7 +55,11 @@
 #'
 #' @export
 assert <- function(data, predicate, ..., success_fun=success_continue,
-                      error_fun=error_stop, title = NULL){
+                      error_fun=error_stop, title = NULL, prediction_must_pass = FALSE){
```

Comment:
After some thought I think we should change the way of thinking about this. Failing mandatory assertion means something important - that data is deeply corrupted, and that's why we should skip all following assertions. Information `skip_following_rules` sounds like hack and doesn't give me enough information, why should I do it.

Let's rename the following parameters:
`prediction_must_pass` --> `mark_data_corrupted_on_failure`
`skip_following_rules` --> `is_corrupted`

`attr(data, "is_corrupted") == TRUE` will mean we can't run the assertion.
Moreover,  we should introduce `skip_fun` in addition to `success_fun` and `error_fun`. This will keep consistency. This is bigger change than just renaming things, so let me know if you have questions about the idea